specification/ml/_types/TrainedModel.ts

```diff
@@ -59,15 +61,16 @@ export class TrainedModelStats {
 }
 
 export class TrainedModelDeploymentStats {
+  adaptive_allocations?: AdaptiveAllocationsSettings
   /** The detailed allocation status for the deployment. */
   allocation_status: TrainedModelDeploymentAllocationStatus
```

Comment:
Per https://github.com/elastic/elasticsearch/blob/045f6a31f994f51d87a217be60251e060132c8a1/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java#L604-L606:

```suggestion
  allocation_status?: TrainedModelDeploymentAllocationStatus
```